File: src/main/java/com/ids1024/whitakerswords/SearchAdapter.java
package com.ids1024.whitakerswords;

import java.util.ArrayList;
import android.widget.TextView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.support.v7.widget.RecyclerView;
import android.text.SpannableStringBuilder;

public class SearchAdapter extends RecyclerView.Adapter<SearchAdapter.ViewHolder> {
    private ArrayList<SpannableStringBuilder> results;

    SearchAdapter(ArrayList<SpannableStringBuilder> results) {
        this.results = results;
    }

    @Override
    public int getItemCount() {
        return results.size();
    }
    
    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.text_view.setText(results.get(position));
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.result, null);
        return new ViewHolder(view);
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        protected TextView text_view;

        public ViewHolder(View view) {
            super(view);
            this.text_view = (TextView)view.findViewById(R.id.result_text);
        }
    }
}

File: src/main/java/com/ids1024/whitakerswords/SearchAdapter.kt
package com.ids1024.whitakerswords

import android.text.SpannableStringBuilder
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import java.util.ArrayList

class SearchAdapter(private var results: ArrayList<SpannableStringBuilder>) : RecyclerView.Adapter<SearchAdapter.ViewHolder>() {

    override fun getItemCount(): Int {
        return results.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.text_view.text = results[position]
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.result, parent, false)
        return ViewHolder(view)
    }

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val text_view: TextView = view.findViewById(R.id.result_text)
    }
}
